static/app/views/performance/landing/widgets/transforms/transformMetricsToVitalList.tsx

```diff
@@ -0,0 +1,68 @@
+import {getParams} from 'sentry/components/organizations/globalSelectionHeader/getParams';
+import {defined} from 'sentry/utils';
+import {MetricsRequestRenderProps} from 'sentry/utils/metrics/metricsRequest';
+import {DEFAULT_STATS_PERIOD} from 'sentry/views/performance/data';
+
+import {QueryDefinitionWithKey, WidgetDataConstraint, WidgetPropUnion} from '../types';
+
+export type VitalsMetricsItem = {
+  transaction: string;
+  measurement_rating: {
+    poor: number;
+    meh: number;
+    good: number;
+  };
+};
+
+export function transformMetricsToVitalList<T extends WidgetDataConstraint>(
+  widgetProps: WidgetPropUnion<T>,
+  results: MetricsRequestRenderProps,
+  _: QueryDefinitionWithKey<T>
+) {
+  const {start, end, utc, interval, statsPeriod} = getParams(widgetProps.location.query, {
+    defaultStatsPeriod: DEFAULT_STATS_PERIOD,
+  });
+
+  const metricsField = widgetProps.Queries.list.fields[0];
+
+  const {errored, loading, reloading, response} = results;
+
+  const data =
+    results.response?.groups.reduce((acc, group) => {
+      const foundTransaction = acc.find(
+        item => item.transaction === group.by.transaction
+      );
+      if (foundTransaction) {
+        foundTransaction.measurement_rating = {
+          ...foundTransaction.measurement_rating,
+          [group.by.measurement_rating]: group.totals[metricsField],
+        };
+        return acc;
+      }
+
+      acc.push({
+        transaction: String(group.by.transaction),
+        measurement_rating: {
+          [group.by.measurement_rating]: group.totals[metricsField],
+        } as VitalsMetricsItem['measurement_rating'],
+      });
+      return acc;
+    }, [] as VitalsMetricsItem[]) ?? [];
+
+  const childData = {
+    loading,
+    reloading,
+    isLoading: loading || reloading,
```

Comment:
do we need `loading` and `isLoading`?

Reply:
yes, all transforms are returning both.
loading is being passed downstream to the `_VitalChart` and isLoading is a part of the `WidgetDataResult` type signature.
this could be refactored into one, but I wanted to stay as consistent as possible with other transforms